### PR TITLE
feat(web): add multi-feature CVE analysis endpoint

### PR DESCRIPTION
**Pull request by kdudka**
## What

Add a new `POST /api/v1/analysis/cve` endpoint that lets OSIM users request multiple CVE feature suggestions in a single HTTP request, supporting the "Full Flaw Suggestions" button (AEGIS-455). The endpoint runs requested features concurrently via `asyncio.gather` with per-feature error isolation — a failing feature returns its error in the `errors` dict while other features still succeed.

## Why

OSIM currently issues a separate API call for each CVE suggestion (impact, CWE, description, etc.). This adds latency and complexity on the client side. A single endpoint that fans out concurrently reduces round-trips and simplifies the frontend integration.

## How to Test

```bash
make all

# manual smoke test (with a running server)
curl -s -X POST http://localhost:9000/api/v1/analysis/cve \
  -H 'Content-Type: application/json' \
  -d '{
    "cve_id": "CVE-2024-1234",
    "title": "kernel: buffer overflow in net subsystem",
    "comment_zero": "A buffer overflow was found in the Linux kernel."
  }' | python3 -m json.tool

# request specific features only
curl -s -X POST http://localhost:9000/api/v1/analysis/cve \
  -H 'Content-Type: application/json' \
  -d '{
    "cve_id": "CVE-2024-1234",
    "features": ["suggest-impact", "suggest-cwe"]
  }' | python3 -m json.tool
```

## Related Tickets

Related: https://redhat.atlassian.net/browse/AEGIS-456
Related: https://redhat.atlassian.net/browse/AEGIS-455

CI: skip-pr-evals

## Summary by Sourcery

Add a multi-feature CVE analysis POST endpoint that runs multiple CVE suggestion features in one request with per-feature error reporting and shared component inference logic.

New Features:
- Introduce POST /api/v1/analysis/cve endpoint to request multiple CVE feature analyses in a single call with optional detail output.
- Define CVEMultiAnalysisRequest and CVEMultiAnalysisResponse models to structure multi-feature CVE analysis requests and results, including error metadata per feature.

Enhancements:
- Centralize CVE component inference into a reusable helper to avoid duplication between single- and multi-feature CVE analysis paths.
- Improve OSIDB CVE retrieval tooling to better leverage and override static context while minimizing unnecessary OSIDB API calls.
- Adjust SuggestAffectedComponents to only fetch OSIDB data directly when the RHFeatureAgent is in use.

Documentation:
- Extend OpenAPI specification with schemas and documentation for the new multi-feature CVE analysis endpoint and its request/response types.

Tests:
- Add test coverage for the multi-feature CVE analysis endpoint, including defaults, failure isolation, component inference behavior, and validation of feature names and required fields.

## Summary by Sourcery

Add a multi-feature CVE analysis POST endpoint that runs multiple CVE suggestion features in a single request with shared component inference and per-feature error reporting.

New Features:
- Introduce a POST /api/v1/analysis/cve endpoint that accepts a CVEMultiAnalysisRequest and returns a CVEMultiAnalysisResponse mapping feature names to results and errors.
- Define FeatureError, CVEMultiAnalysisRequest, and CVEMultiAnalysisResponse models for structured multi-feature CVE analysis requests and per-feature error metadata.

Enhancements:
- Centralize CVE component inference into a reusable helper shared by single- and multi-feature CVE analysis paths to avoid duplication.
- Improve OSIDB CVE retrieval logic to better leverage and selectively override static context while minimizing unnecessary OSIDB API calls.
- Adjust SuggestAffectedComponents to only fetch OSIDB data directly when the RHFeatureAgent is used.

Documentation:
- Extend the OpenAPI specification and schemas to document the new multi-feature CVE analysis endpoint and its request/response types, including FeatureError.

Tests:
- Add comprehensive tests for the multi-feature CVE analysis endpoint covering defaults, error isolation, component inference behavior, and validation of feature names and required fields.

### CODE REVIEW

File: src/aegis_ai_web/src/main.py [hunk 1]
```diff
@@ -363,6 +427,57 @@ async def cve_analysis(feature: CVEFeatureName, cve_id: CVEID, detail: bool = Fa
...
+                status_code=422,
+                detail=f"Unknown feature(s): {invalid}. Valid features: {valid}",
+            )
+
+    validated_input: dict[str, Any] = {"cve_id": cve_id}
+    if request_body.model_extra:
+        validated_input.update(request_body.model_extra)
+
+    skip_inference = all(f == "suggest-affected-components" for f in requested_features)
+    enriched_input = await _maybe_infer_components(
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/src/main.py]: Fixed in ef5b1300b5251976a3b8a79854bd55eff9b646e3.

File: src/aegis_ai_web/src/main.py [hunk 2]
```diff
@@ -363,6 +427,57 @@ async def cve_analysis(feature: CVEFeatureName, cve_id: CVEID, detail: bool = Fa
...
+        cve_id, validated_input, skip_inference=skip_inference
+    )
+
+    async def _run_feature(feature_name: str) -> tuple[str, Any]:
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/src/main.py]: Fixed in ef5b1300b5251976a3b8a79854bd55eff9b646e3.

File: docs/openapi.yml [hunk 3]
```diff
@@ -330,6 +372,41 @@ components:
...
+            items:
+              type: string
+          - type: 'null'
+          title: Features
+          description: Feature names to run. If omitted or empty, all features are run.
+      type: object
+      required:
+      - cve_id
+      additionalProperties: true
+      title: CVEMultiAnalysisRequest
```
[the remark this answers is not in the document]
Comment (kdudka) [reply docs/openapi.yml]: Fixed.

File: src/aegis_ai_web/tests/test_endpoints.py [hunk 4]
```diff
@@ -321,3 +321,217 @@ def test_suggest_affected_components_post_endpoint():
...
+    def test_osidb_error_mapped_per_feature(self):
+        mock_result = _make_mock_result(_make_sac_output())
+        with _patch_all_cve_feature_execs(mock_result):
+            with patch.object(
+                cve_features.SuggestCWE,
+                "exec",
+                new_callable=AsyncMock,
+                side_effect=OSIDBFlawNotFoundError("CVE-2024-1234"),
+            ):
+                response = client.post(
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/tests/test_endpoints.py:450]: Sounds like overkill for our needs.  The biggest user of this API is OSIM, which does not even need OSIDB auth on the Aegis backend.

File: src/aegis_ai_web/tests/test_endpoints.py [hunk 5]
```diff
@@ -321,3 +321,217 @@ def test_suggest_affected_components_post_endpoint():
...
+    def test_features_omitted_runs_defaults(self):
+        mock_result = _make_mock_result(_make_sac_output())
+        with _patch_all_cve_feature_execs(mock_result):
+            response = client.post(
+                "/api/v1/analysis/cve",
+                json={"cve_id": "CVE-2024-1234"},
+            )
+        assert response.status_code == 200
+        data = response.json()
+        assert set(data["results"].keys()) == set(DEFAULT_CVE_FEATURES)
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/tests/test_endpoints.py:400]: Fixed in e8995a6ab247d1cb0a4a945b03e68f7ba0902970.

File: src/aegis_ai_web/tests/test_endpoints.py [hunk 6]
```diff
@@ -321,3 +321,217 @@ def test_suggest_affected_components_post_endpoint():
...
+    def test_component_inference_called_once(self):
+        mock_result = _make_mock_result(_make_sac_output())
+        sac_result = _make_mock_result(_make_sac_output())
+
+        with _patch_all_cve_feature_execs(mock_result):
+            with patch.object(
+                cve_features.SuggestAffectedComponents,
+                "exec",
+                new_callable=AsyncMock,
+                return_value=sac_result,
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/tests/test_endpoints.py:491]: This is a corner case really.  API for components suggestion was available before this feature was introduced.

File: src/aegis_ai_web/src/main.py [hunk 7]
```diff
@@ -315,13 +322,70 @@ async def generate_response(
...
+) -> dict:
+    """Conditionally run SuggestAffectedComponents and return enriched context."""
+    if skip_inference:
+        return validated_input
+
+    existing_components = validated_input.get("components") or []
+    description_text = (
+        validated_input.get("comment_zero") or validated_input.get("cve_description")
+    ) or ""
+    if not existing_components and validated_input.get("title") and description_text:
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/src/main.py:355]: I do not understand.  If `title` and `description` is available but not `components`, `SuggestAffectedComponents` should be executed.

File: src/aegis_ai/toolsets/tools/osidb/__init__.py [hunk 8]
```diff
@@ -93,6 +93,34 @@ class CVE(BaseToolOutput):
...
+    overrides: dict[str, Any] = {}
+    for field in _CVE_DATA_FIELDS:
+        value = ctx.get(field)
+        # OSIM sends "cve_description" instead of "description"
+        if not value and field == "description":
+            value = ctx.get("cve_description")
+        if value:
+            overrides[field] = value
+    if overrides:
+        return cve.model_copy(update=overrides)
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai/toolsets/tools/osidb/__init__.py]: Fixed.

File: src/aegis_ai_web/src/main.py [hunk 9]
```diff
@@ -363,6 +427,57 @@ async def cve_analysis(feature: CVEFeatureName, cve_id: CVEID, detail: bool = Fa
...
+    )
+
+    async def _run_feature(feature_name: str) -> tuple[str, Any]:
+        FeatureClass = cve_feature_registry[feature_name]
+        feature_instance = FeatureClass(agent=llm_agent)
+        result = await feature_instance.exec(cve_id, static_context=enriched_input)
+        return result if detail else result.output
+
+    tasks = [_run_feature(name) for name in requested_features]
+    outcomes = await asyncio.gather(*tasks, return_exceptions=True)
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/src/main.py:501]: Fixed in 8fb4242d5790a7c2ddf90b36e5297542f00e49db.

File: src/aegis_ai_web/src/main.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/src/main.py:355]: Got it.  Fixed in 4497b0fccf0582ad643fb4a6e634b3927934ef27.

File: src/aegis_ai_web/tests/test_endpoints.py [hunk 10]
```diff
@@ -321,3 +321,221 @@ def test_suggest_affected_components_post_endpoint():
...
+    def test_invalid_feature_name(self):
+        response = client.post(
+            "/api/v1/analysis/cve",
+            json={
+                "cve_id": "CVE-2024-1234",
+                "features": ["nonexistent-feature"],
+            },
+        )
+        assert response.status_code == 422
+        assert "nonexistent-feature" in response.json()["detail"]
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/tests/test_endpoints.py:473]: This is a corner case really.  I do not want to maintain test coverage for this.

File: src/aegis_ai_web/tests/test_endpoints.py [hunk 11]
```diff
@@ -321,3 +321,221 @@ def test_suggest_affected_components_post_endpoint():
...
+    def test_osidb_error_mapped_per_feature(self):
+        mock_result = _make_mock_result(_make_sac_output())
+        with _patch_all_cve_feature_execs(mock_result):
+            with patch.object(
+                cve_features.SuggestCWE,
+                "exec",
+                new_callable=AsyncMock,
+                side_effect=OSIDBFlawNotFoundError("CVE-2024-1234"),
+            ):
+                response = client.post(
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/tests/test_endpoints.py:450]: Sounds like overkill for our needs. The biggest user of this API is OSIM, which does not even need OSIDB auth on the Aegis backend.

File: docs/openapi.yml [hunk 12]
```diff
@@ -330,6 +372,41 @@ components:
...
+            items:
+              type: string
+          - type: 'null'
+          title: Features
+          description: Feature names to run. If omitted or empty, a default set of stable features is run.
+      type: object
+      required:
+      - cve_id
+      additionalProperties: true
+      title: CVEMultiAnalysisRequest
```
[the remark this answers is not in the document]
Comment (kdudka) [reply docs/openapi.yml:394]: This is not a bottleneck of the API really.  The API could be easily overloaded in gazillion other ways.

File: src/aegis_ai_web/src/main.py [hunk 13]
```diff
@@ -315,13 +322,72 @@ async def generate_response(
...
+    if isinstance(exc, OSIDBUnauthorizedError):
+        return FeatureError(error="OSIDBUnauthorizedError", detail=str(exc))
+    if isinstance(exc, OSIDBAuthError):
+        return FeatureError(error="OSIDBAuthError", detail=str(exc))
+    log_exception_safely(exc, f"Error executing CVE feature '{feature_name}'")
+    return FeatureError(
+        error=type(exc).__name__,
+        detail=f"An internal error occurred while executing CVE feature '{feature_name}'.",
+    )
+
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/src/main.py:397]: Good point.  Fixed in e46507c826ae88ac666b5edd6c0fa11f1d7f02fa.

File: src/aegis_ai_web/src/main.py [hunk 14]
```diff
@@ -363,6 +437,60 @@ async def cve_analysis(feature: CVEFeatureName, cve_id: CVEID, detail: bool = Fa
...
+    async def _run_feature(feature_name: str) -> Any:
+        FeatureClass = cve_feature_registry[feature_name]
+        feature_instance = FeatureClass(agent=llm_agent)
+        result = await feature_instance.exec(cve_id, static_context=enriched_input)
+        return result if detail else result.output
+
+    tasks = [_run_feature(name) for name in requested_features]
+    outcomes = await asyncio.gather(*tasks, return_exceptions=True)
```
[the remark this answers is not in the document]
Comment (kdudka) [reply src/aegis_ai_web/src/main.py]: Fixed in 8bfb6708affd98b49800e6b542fc0729d4fb7015.

**Review by superbuggy — APPROVED** [review]
